rholang/src/main/scala/coop/rchain/rholang/interpreter/implicits.scala

```diff
@@ -200,16 +200,51 @@ object implicits {
         that.freeCount + p.freeCount,
         that.locallyFree | p.locallyFree
       )
+
+    // TODO: Receives and matches
+    def wildcard: Boolean =
+      p.sends.map(send => SendLocallyFree.wildcard(send)).exists(identity) || p.evals
+        .map(eval => EvalLocallyFree.wildcard(eval))
+        .exists(identity) || p.exprs.map(expr => ExprLocallyFree.wildcard(expr)).exists(identity)
```

Comment:
The fact the the best way is to use exists(identity) is a giant red mark against scala.

Reply:
Hmm what is the Haskell way to do this?

Also I'm not sure if this is the way its supposed to be in Scala. Expert opinions @henrytill ?

Reply:
To give some context, I'm folding over a list of booleans using OR .